api/ocm/plugin/README.md

```diff
@@ -0,0 +1,151 @@
+	"ocm.software/ocm/cmds/demoplugin/config"
+	"ocm.software/ocm/cmds/demoplugin/uploaders"
+	"ocm.software/ocm/cmds/demoplugin/valuesets"
+)
+
+func main() {
+	p := ppi.NewPlugin("demo", version.Get().String())
+
+	p.SetShort("demo plugin")
+	p.SetLong("plugin providing access to temp files and a check routing slip entry.")
+	p.SetConfigParser(config.GetConfig)
+
+	p.RegisterAccessMethod(accessmethods.New())
+	u := uploaders.New()
+	p.RegisterUploader("testArtifact", "", u)
+	p.RegisterValueSet(valuesets.New())
+	err := cmds.NewPluginCommand(p).Execute(os.Args[1:])
+	if err != nil {
+		os.Exit(1)
+	}
+}
+```
+
+The library provides an interface type [Plugin](ppi/interface.go) and a standard [implementation](ppi/plugin.go) for this interface.
+
+It is used to configure the desired extension point implementations. Every extension point provides an extension interface and supporting types.
+By implementing and registering such an interface the plugin object gets
+informed about the implemented extension point variants and how they are implemented.
+
+This information is then used by the standard implementation of the `info` command
+to provide an appropriate plugin descriptor.
+
+The support provides a complete command set. Therefore, the extension point implementation have never to deal with the command line interface, everything is described by the provided extension point interfaces.
+
+The implemented standard commands have access to the plugin object and therefore
+determine whether there is an implementation for an extension point variant requested via the command execution.
+
+If there is an implementation, the CLI interface is just mapped to calls to interface functions of the particular extension point functionality.
+
+The standard implementation od the commands can be found below the package
+[`api/plugin/ppi/cmds](ppi/cmds), structured by the extension point name and its
+interface operation. (for example [upload/put](ppi/cmds/upload/put/cmd.go))
+
+## How plugins are used in the library
+
+The library includes the counterpart of the plugin-side support. It is found directly in package [`api/ocm/plugin](.). It shared the descriptor package for the plugin descriptor and provides a separate [Plugin](plugin.go) object type.
+
+Such an object provides methods for all extension point functions, which can the appropriate CLI command for its plugin executable. Again, it shields the CLI interface by providing an appropriate Go interface.
+
+To enable the usage of plugins in an [api/ocm/Context](../internal/context.go),
+the plugins must be registered at this context by calling [api/ocm/plugin/registration/RegisterExtensions](registration/registration.go).
+It used the plugin cache die attribute to setup a plugin cache by reading all the plugin descriptors of the configured plugins and providing appropriate plugin objects.
+
+For extension points requiring a static registration (like access method) it uses
```

Comment:
What is the difference between a static and dynamic registration? this is not covered in this document. And when do you require what? 
Also: How does a CLI maintainer write a new extension point?